code/game/machinery/newscaster.dm

```diff
@@ -887,6 +887,16 @@ GLOBAL_LIST_EMPTY(allCasters)
 	var/wantedBody
 	var/wantedPhoto
 	var/creationTime
+	var/datum/horoscope/horoscope
+
+/obj/item/newspaper/Initialize(mapload)
+	. = ..()
+	horoscope = new
+
+/obj/item/newspaper/examine(mob/user)
+	. = ..()
+	if(user && !(REF(user) in horoscope.has_read))
```

Comment:
don't hide this if they read their horoscope, they can still re-read it, there's no reason in removing the information after they read it